core/src/main/java/org/apache/calcite/materialize/MaterializationService.java

```diff
@@ -126,6 +126,10 @@ public MaterializationKey defineMaterialization(final CalciteSchema schema,
     if (tableEntry == null) {
       tableEntry = schema.getTableBySql(viewSql);
     }
+    if (tableEntry == null) {
```

Comment:
Can we use the {{existing}} flag as we do for pre-populated materialization table, and move this if block into the {{if (existing)}} block?

Reply:
Sounds like a good idea.